src/detect-parse.h

```diff
@@ -39,6 +39,21 @@ enum {
     SIG_DIREC_DST
 };
 
+typedef struct DetectParseRegex_ {
+    pcre *regex;
+    pcre_extra *study;
+    pcre_jit_stack *jit_stack;
+    struct DetectParseRegex_ *next;
+} DetectParseRegex;
+
+#if PCRE_HAVE_JIT
+#define PCRE_EXEC(p, str, x1, x2, ss, ss_cnt) \
+        pcre_jit_exec((p)->regex, (p)->study, str, strlen(str), x1, x2, ss, ss_cnt, (p)->jit_stack)
+#else
+#define PCRE_EXEC(p, str, x1, x2, ss, ss_cnt) \
```

Comment:
Is there a reason for making this a macro instead of a function?

Reply:
Given your comments about using JIT only in `detect-pcre.c`, I plan to remove the macro but will retain the "cleanup" related changes in the detect modules.

I'll ensure that only uses from `detect-pcre.c` enable/allocate/assign the JIT stack as well.

Will your OS changes (e.g., result of `PageSupportsRWX` at https://github.com/OISF/suricata/blob/master/src/detect-pcre.c#L147) need to be retained?

Reply:
Probably, yes.